### PR TITLE
feat(php8): promoted params v2

### PR DESCRIPTION
**Pull request by cseufert**
This patch is more inline with nikic PHP-Parser AST syntax, and should be 100% backward compatible with existing ast consumers.

### CODE REVIEW

File: src/parser/function.js [hunk 1]
```diff
@@ -244,6 +253,23 @@ module.exports = {
...
+      this.next();
+      return MODIFIER_PUBLIC;
+    } else if (this.token === this.tok.T_PROTECTED) {
+      this.next();
+      return MODIFIER_PROTECTED;
+    } else if (this.token === this.tok.T_PRIVATE) {
+      this.next();
+      return MODIFIER_PRIVATE;
+    } else {
+      0;
```
Comment (czosel) [inline src/parser/function.js:270]: Should this be `return 0;`?

**Comment by czosel** [comment]
Looks great otherwise :+1:

File: src/parser/function.js (hunk 1 above)
Comment (cseufert) [reply src/parser/function.js]: Thanks, i missed that

**Comment by czosel** [comment]
Looks like you’ll still need to run prettier to make the linter happy..

**Comment by czosel** [comment]
I went ahead and did the formatting fix myself in #702

**Pull request merged**